### PR TITLE
harfbuzz: add version 7.0.1, use official tarballs

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **harfbuzz/***

- add version 7.0.1
- use official tarballs
- sort config.yml, conandata.yml by versions
- disable util build

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/harfbuzz/all/conanfile.py [hunk 1]
```diff
@@ -74,30 +75,37 @@ def configure(self):
...
+            self.requires("freetype/2.12.1")
+        if self.options.with_icu:
+            self.requires("icu/72.1")
+        if self.options.with_glib:
+            self.requires("glib/2.75.2")
+
+    def package_id(self):
+        if self.options.with_glib and not self.options["glib"].shared:
+            self.info.requires["glib"].full_package_mode()
+
```
Comment (SpaceIm) [inline recipes/harfbuzz/all/conanfile.py:92]:
```suggestion
```

https://github.com/conan-io/conan-center-index/pull/16018

Comment (toge) [reply recipes/harfbuzz/all/conanfile.py:92]: @SpaceIm 
Thanks!

**Review by SpaceIm — APPROVED** [review]

File: recipes/harfbuzz/config.yml [hunk 2]
```diff
@@ -1,15 +1,17 @@
 versions:
-  "3.2.0":
+  "7.0.0":
```
Comment (uilianries) [inline recipes/harfbuzz/config.yml:2]:
```suggestion
  "7.0.1":
```

**Review by uilianries — APPROVED** [review]

**Comment by SpaceIm** [comment]
where is v2 pipeline result?

**Comment by uilianries** [comment]
It's still running, but Conan v1 didn't wait v2 because it was triggered internally 😞

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/harfbuzz/all/conanfile.py [hunk 3]
```diff
@@ -78,6 +78,9 @@ def requirements(self):
         if self.options.with_glib:
             self.requires("glib/2.76.0", run=can_run(self))
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def validate(self):
```
Comment (prince-chrismc) [inline recipes/harfbuzz/all/conanfile.py:84]:
```suggestion
    def validate(self):
```

File: recipes/harfbuzz/all/conanfile.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 from conan.tools.apple import is_apple_os, fix_apple_shared_install_name
 from conan.tools.build import can_run, stdcpp_library
 from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
-from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rm, rmdir
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rm, rmdir, rename, replace_in_file
```
Comment (prince-chrismc) [inline recipes/harfbuzz/all/conanfile.py:6]:
```suggestion
from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rm, rmdir, replace_in_file
```

**Comment by prince-chrismc** [comment]
I did my best to fixup the merge conflicts

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/harfbuzz/all/conanfile.py [hunk 5]
```diff
@@ -184,6 +185,7 @@ def package_info(self):
             if libcxx:
                 self.cpp_info.system_libs.append(libcxx)
 
+
```
Comment (SpaceIm) [inline recipes/harfbuzz/all/conanfile.py:188]:
```suggestion
```

and c3i build has to be triggered anyway.

Comment (prince-chrismc) [reply recipes/harfbuzz/all/conanfile.py:188]: I dont want to wait for reviews since it's approved, but yes ideally

**Comment by prince-chrismc** [comment]
This was exact same time as https://github.com/conan-io/conan-center-index/pull/16725#issuecomment-1493040940 :/

**Pull request merged**